src/tests/ftest/io/dfuse_sparse_file.py

```diff
@@ -0,0 +1,141 @@
+        # remote node
+        ssh = paramiko.SSHClient()
+        ssh.load_system_host_keys()
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        ssh.connect(self.hostlist_clients[0], username=getuser())
+        sftp = ssh.open_sftp()
+
+        # open remote file
+        file_obj = sftp.open(sparse_file, 'r+')
+        # set file size to max available nvme size
+        sftp.truncate(sparse_file, self.space_before)
+        fsize_after_truncate = get_remote_file_size(self.hostlist_clients[0],
+                                                    sparse_file)
+        self.log.info("File size (in bytes) after truncate: %s",
+                      fsize_after_truncate)
+        # verifying the file size got set to desired value
+        self.assertTrue(fsize_after_truncate == self.space_before)
+
+        # write to the first byte of the file with char 'A'
+        dd_first_byte = u"echo 'A' | dd conv=notrunc of={} bs=1 count=1".\
+                        format(sparse_file)
+        self.execute_cmd(dd_first_byte)
+        fsize_write_1stbyte = get_remote_file_size(self.hostlist_clients[0],
+                                                   sparse_file)
+        self.log.info("File size (in bytes) after writing first byte: %s",
+                      fsize_write_1stbyte)
+        # verify file did not got overriten after dd write.
+        self.assertTrue(fsize_write_1stbyte == self.space_before)
+
+        # write to the 1024th byte position of the file
+        dd_1024_byte = u"echo 'A' | dd conv=notrunc of={} obs=1 seek=1023 \
+                       bs=1 count=1".format(sparse_file)
+        self.execute_cmd(dd_1024_byte)
+        fsize_write_1024thwrite = get_remote_file_size(self.hostlist_clients[0],
+                                                       sparse_file)
+        self.log.info("File size (in bytes) after writing 1024th byte: %s",
+                      fsize_write_1024thwrite)
+        # verify file did not got overriten after dd write.
+        self.assertTrue(fsize_write_1024thwrite == self.space_before)
+
+        # Obtain the value of 1st byte and 1024th byte in the file and
+        # compare their values, they should be same.
+        check_first_byte = file_obj.read(1)
+        file_obj.seek(1022, 1)
+        check_1024th_byte = file_obj.read(1)
+        self.assertTrue(check_first_byte == check_1024th_byte)
+
+        # check the middle 1022 bytes if they are filled with zeros
+        middle_1022_bytes = u"cmp --ignore-initial=1 --bytes=1022 {} {}".\
+                                  format(sparse_file, "/dev/zero")
+        self.execute_cmd(middle_1022_bytes)
+
+        # read last 512 bytes which should be zeros till end of file.
+        ignore_bytes = self.space_before - 512
+        read_till_eof = u"cmp --ignore-initial={} {} {}".format(
+            ignore_bytes, sparse_file, "/dev/zero")
+        self.execute_cmd(read_till_eof, False)
```

Comment:
I guess you have to set "ret = self.execute_cmd(read_till_eof, False)" and use it in line no. 139.

Reply:
Accidentally forgot to copy the updated file. It should be fine now.